### DISCUSSION TITLE
Empty array as default value in supabase ui

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#9471] [Questions] [answered]
## Problem
I tried to set an empty array as the default value in my table column but it always errors out.
My column is a `VARCHAR[]`.

## Reason
It seems more consistent to always return the same datatype to the frontend, so I'm trying to avoid returning `NULL` when expecting an array.

## What I've tried
Setting:
- []
- ""
- “{}"
- array[]::varchar[]
 
Interestingly it works when setting it via SQL in the console.
Then it shows in the UI with `ARRAY[]::character varying[]` but when setting this value by hand in the UI it won't work.

Setting an empty array on `UPDATE` in the front end works.

**Comment by GaryAustin1** [comment, marked as answer]
try just `{}`.

**Reply by bennik88** [reply]
Thanks, Gary. This works. Do you know why postgres sometimes refers to arrays with `[]` and sometimes with `{}`? I read something about it in the [docs](https://www.postgresql.org/docs/14/arrays.html). But don't understand the syntax, to be honest. Therefore don't know when to use what. I found out that curly braces (`{}`) are used to [stringify something in postgres](https://stackoverflow.com/questions/46751330/postgresql-when-to-use-array-constructor-vs-curly-brackets). Woudn't that mean `{}` is equivalent to `""` or probably to `[""]`.

**Reply by GaryAustin1** [reply]
https://www.postgresql.org/docs/current/arrays.html
See 8.15.2. Array Value Input

**Reply by bennik88** [reply]
That’s the same link, I linked in my last comment with „docs“ I already read but didn’t fully understand. :)

**Reply by GaryAustin1** [reply]
Ha, sorry, did not click "doc".   
Basically in Postgres you can represent an array in a string form like '{"cat","dog"}' .
Or another format where you use the word ARRAY like ARRAY ["cat","dog"].  
I can't comment on why the table UI works the way it does with different formats.

**Reply by bennik88** [reply]
:) no worries. Thanks. Didn’t came across the curly braces with single quotes equals an array syntax yet so appreciate clarifying this.
